### PR TITLE
Update changelog for 4.1.0

### PR DESCRIPTION
**Pull request by theopolis**
This takes advantage of GitHub's display of PRs/issues using hashtag-shorthand.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,52 @@
...
+- Add logger_tls_max_lines flag #5956
+- Add AWS Session Token support #5944
+
+### Build
+
+- Lots of work on CPack-based packaging #5809, #5822, #5823, #5827, #5780, #5850, #5843, #5881, #5825, #5940, #5951, #5936
+- Lots of work porting Python2 to Python3 #5846
+- Use SQLite 3.29.0 on Windows and macOS #5810
+- Use aws-sdk-cpp source-builds on Windows and macOS #5889
+- Add various code quality checks and utilities #5834, #5730, #5872
```
Comment (alessandrogario) [inline CHANGELOG.md]: We could also add that we upgraded OpenSSL from 1.0.2o to version 1.0.2t (https://www.openssl.org/news/cl102.txt) and switched to the formula on all platforms

Comment (alessandrogario) [reply CHANGELOG.md]: Is it normal that there are no links generated for the hashtag shorthands in the preview under "Files changed"?

Comment (theopolis) [reply CHANGELOG.md]: That doesn't look correct :/

**Comment by theopolis** [comment]
@alessandrogario, thanks for the review, I addressed your recommendations.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+- Add various code quality checks and utilities ([#5834](https://github.com/osquery/osquery/pull/5834)), ([#5730](https://github.com/osquery/osquery/pull/5730)), ([#5872](https://github.com/osquery/osquery/pull/5872))
+
+### Harderning
+
+- Restore fuzzing harness and use oss-fuzz ([#5844](https://github.com/osquery/osquery/pull/5844)), ([#5886](https://github.com/osquery/osquery/pull/5886)), ([#5910](https://github.com/osquery/osquery/pull/5910)), ([#5915](https://github.com/osquery/osquery/pull/5915)), ([#5923](https://github.com/osquery/osquery/pull/5923)), ([#5955](https://github.com/osquery/osquery/pull/5955)), ([#5963](https://github.com/osquery/osquery/pull/5963))
+- Use newer RapidJSON and switch to safer iterative parsing ([#5893](https://github.com/osquery/osquery/pull/5893)), ([#5913](https://github.com/osquery/osquery/pull/5913))
+
+### Bug Fixes
+
+- Set Windows MSI ErrorControl to normal instead of critical #8e67884
```
Comment (alessandrogario) [inline CHANGELOG.md]: It seems like this is also not showing up as a link, so we could use something like this instead:

```
[commit 8e67884](https://github.com/osquery/osquery/commit/8e67884)
```

It's also fine as it is for me!

**Pull request merged**